torchgeo/datasets/soda.py

```diff
@@ -0,0 +1,363 @@
+        },
+    }
+
+    classes = (
+        'airplane',
+        'helicopter',
+        'small-vehicle',
+        'large-vehicle',
+        'ship',
+        'container',
+        'storage-tank',
+        'swimming-pool',
+        'windmill',
+        'other',
+    )
+
+    valid_splits = ('train', 'val', 'test')
+
+    valid_orientations = ('oriented', 'horizontal')
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        split: Literal['train', 'val', 'test'] = 'train',
+        bbox_orientation: Literal['oriented', 'horizontal'] = 'horizontal',
+        transforms: Callable[[dict[str, Tensor]], dict[str, Tensor]] | None = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new instance of SODA-A dataset.
+
+        Args:
+            root: root directory where dataset can be found
+            split: one of "train", "val", or "test"
+            bbox_orientation: one of "oriented" or "horizontal"
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            AssertionError: if *split* or *bbox_orientation* argument is invalid
+            DatasetNotFoundError: If dataset is not found and *download* is False.
+        """
+        assert split in self.valid_splits, f'split must be one of {self.valid_splits}'
+
+        assert bbox_orientation in self.valid_orientations, (
+            f'bbox_orientation must be one of {self.valid_orientations}'
+        )
+
+        self.root = root
+        self.split = split
+        self.bbox_orientation = bbox_orientation
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+
+        self._verify()
+
+        self.sample_df = pd.read_csv(os.path.join(self.root, 'sample_df.csv'))
+        self.sample_df = self.sample_df[
+            self.sample_df['split'] == self.split
+        ].reset_index(drop=True)
+
+    def __len__(self) -> int:
+        """Return the number of samples in the dataset."""
+        return len(self.sample_df)
+
+    def __getitem__(self, idx: int) -> dict[str, Tensor]:
+        """Return the sample at the given index.
+
+        Args:
+            idx: index of the sample to return
+
+        Returns:
+            the sample at the given index
+        """
+        row = self.sample_df.iloc[idx]
+
+        image = self._load_image(os.path.join(self.root, row['image_path']))
+        boxes, labels = self._load_labels(os.path.join(self.root, row['label_path']))
+
+        sample: dict[str, Tensor] = {'image': image, 'label': labels}
+
+        if self.bbox_orientation == 'oriented':
+            sample['boxes'] = boxes
+        else:
+            sample['boxes_xyxy'] = boxes
```

Comment:
```suggestion
            sample['bbox'] = boxes
        else:
            sample['bbox_xyxy'] = boxes
```
See https://github.com/kornia/kornia/blob/main/kornia/constants.py#L142 for valid keys. Since rotated bboxes aren't yet supported, let's just call it bbox until we add support for that.